### PR TITLE
Update mx to 5.8.0 and fix checkstyle issues

### PR DESCRIPTION
**Pull request by smarr**
This fixes the superfluous keyword issues reported by the latest checkstyle version and updates mx to make sure these rules are enforced in the future.

### CODE REVIEW

File: truffle/com.oracle.truffle.tools/src/com/oracle/truffle/tools/NodeExecCounter.java [hunk 1]
```diff
@@ -121,7 +121,7 @@ public void onEnter(Probe probe, Node node, VirtualFrame frame) {
                 /*
                  * Everything up to here is inlined by Truffle compilation. Delegate the next part
                  * to a method behind an inlining boundary.
-                 *
+                 *
```
Comment (woess) [inline truffle/com.oracle.truffle.tools/src/com/oracle/truffle/tools/NodeExecCounter.java]: (unnecessary whitespace change)

**Comment by woess** [comment]
looks good. feel free to push.

File: truffle/com.oracle.truffle.tools/src/com/oracle/truffle/tools/NodeExecCounter.java (hunk 1 above)
Comment (smarr) [inline truffle/com.oracle.truffle.tools/src/com/oracle/truffle/tools/NodeExecCounter.java]: sorry, saw the comment only after pushing

Comment (woess) [inline truffle/com.oracle.truffle.tools/src/com/oracle/truffle/tools/NodeExecCounter.java]: it's ok
